src/test/java/redis/clients/jedis/JedisPooledClientSideCacheTest.java

```diff
@@ -0,0 +1,100 @@
+package redis.clients.jedis;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.function.Supplier;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class JedisPooledClientSideCacheTest {
+
+  protected static final HostAndPort hnp = HostAndPorts.getRedisServers().get(1);
+
+  protected Jedis control;
+
+  @Before
+  public void setUp() throws Exception {
+    control = new Jedis(hnp, DefaultJedisClientConfig.builder().password("foobared").build());
+    control.flushAll();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    control.close();
+  }
+
+  private static final Supplier<JedisClientConfig> clientConfig
+      = () -> DefaultJedisClientConfig.builder().resp3().password("foobared").build();
+
+  private static final Supplier<GenericObjectPoolConfig<Connection>> singleConnectionPoolConfig
+      = () -> {
+        ConnectionPoolConfig poolConfig = new ConnectionPoolConfig();
+        poolConfig.setMaxTotal(1);
+        return poolConfig;
+      };
+
+  @Test
+  public void simple() {
+    try (JedisPooled jedis = new JedisPooled(hnp, clientConfig.get(), new ClientSideCache())) {
+      control.set("foo", "bar");
+      assertEquals("bar", jedis.get("foo"));
+      control.del("foo");
+      assertThat(jedis.get("foo"), Matchers.oneOf("bar", null)); // ?
+    }
+  }
+
+  @Test
+  public void simpleWithSimpleMap() {
```

Comment:
Can we change this to use @ParameterizedTest, like other clients. The idea is that your test itself becomes reusable code, and the inputs are the validation piece. WDYT?

Simple unrelated example:

```@ParameterizedTest
@ValueSource(ints = {1, 3, 5, -3, 15, Integer.MAX_VALUE}) // six numbers
void isOdd_ShouldReturnTrueForOddNumbers(int number) {
    assertTrue(Numbers.isOdd(number));
}
```

Reply:
Good suggestion.
Needs a lot of time and effort (for the repo).
I don't see as relevant to this specific test.